weave/engine_trace.py

```diff
@@ -329,3 +328,15 @@ def statsd():
 
 def datadog_is_enabled():
     return os.getenv("DD_ENV")
+
+
+def set_pii_metric(span, key, val, pii_val):
+    span.set_metric(key, pii_val) if os.getenv(
+        "DISABLE_WEAVE_PII"
+    ) else span.set_metric(key, val)
+
+
+def set_pii_tag(span, key, val, pii_val):
```

Comment:
Hey Josiah - I think there are some improvements that can be made here. This change still requires the developer to remember to make these calls when needed. More importantly, they can still just as easily call span.set_tag directly. I think it would be a bit better if `tracer.trace` returned a span object that had a `set_metric` method that accepts 3 params: `set_metric(name, value, pii_redacted = false)` ... this way the payload is logged in saas always, but only logged in server when pii_redacted has been explicitly set to true. Also, there is no other way for the dev to log metrics since the underlying DD span is hidden from them.